src/Symfony/Component/Config/Builder/ConfigBuilderGenerator.php

```diff
@@ -469,10 +510,24 @@ private function getSingularName(PrototypedArrayNode $node): string
         return $name;
     }
 
-    private function buildToArray(ClassBuilder $class): void
+    private function buildToArray(ClassBuilder $class, bool $rootClass = false): void
     {
-        $body = '$output = [];';
+        $body = '';
+        if ($rootClass) {
+            $body = 'if ($this->configOutput) {
```

Comment:
Maybe use nowdoc syntax for readability? This also unlocks highlighting in some IDEs (PhpStorm) if the identifier is the language name.

Reply:
Unfortunately not possible here, because we need some spacing for `$output` to be added in the case of `$rootClass` being true. With nowdoc syntax, the four spaces are automatically removed because there's not character after. But I did the update at your other suggestion's place 🙂